pkg/deploy/api/validation/validation.go

```diff
@@ -39,7 +39,18 @@ func ValidateDeploymentConfigSpec(spec deployapi.DeploymentConfigSpec) field.Err
 	if spec.Template == nil {
 		allErrs = append(allErrs, field.Required(specPath.Child("template"), ""))
 	} else {
+		// rather than handleEmptyImageReferences returning an array of changed containers, PR review deemed it more
+		// efficient to build a list of the original container to image mappings, and then just restore all of them (and hence,
+		// even the ones not modified)
+		originalContainerImages := make(map[string]string)
+		for _, container := range spec.Template.Spec.Containers {
+			originalContainerImages[container.Name] = container.Image
+		}
+		handleEmptyImageReferences(spec.Template, spec.Triggers)
```

Comment:
He said both deep copy and array return were overkill.  I'll try to ref the
comment:  https://github.com/openshift/origin/pull/9167#discussion_r66316565

But just in case, an old fashioned copy / paste:

"It would be better to have a map of container name to original image and
just set them all. Returning array could be overkill."

It came after the comment you referenced.

On Wed, Jun 8, 2016 at 4:33 PM, Jordan Liggitt <notifications@github.com>
wrote:

> In pkg/deploy/api/validation/validation.go
> <https://github.com/openshift/origin/pull/9167#discussion_r66334076>:
>
> > @@ -39,7 +39,18 @@ func ValidateDeploymentConfigSpec(spec deployapi.DeploymentConfigSpec) field.Err
> >  	if spec.Template == nil {
> >  		allErrs = append(allErrs, field.Required(specPath.Child("template"), ""))
> >  	} else {
> > +		// rather than handleEmptyImageReferences returning an array of changed containers, PR review deemed it more
> > +		// efficient to build a list of the original container to image mappings, and then just restore all of them (and hence,
> > +		// even the ones not modified)
> > +		originalContainerImages := make(map[string]string)
> > +		for _, container := range spec.Template.Spec.Containers {
> > +			originalContainerImages[container.Name] = container.Image
> > +		}
> > +		handleEmptyImageReferences(spec.Template, spec.Triggers)
>
> I think he said deep copy was overkill... I think what I'm proposing is
> similar to what he suggested in #9167 (comment)
> <https://github.com/openshift/origin/pull/9167#discussion_r65827639>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/9167/files/abd1b6fed22e427f856c6ac25b536ec3ce45b70f#r66334076>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ADbadHNAdUUVb5A9HGiCeJj7NZnq_Q1Nks5qJycPgaJpZM4It4tM>
> .
>